### PR TITLE
refactor to use go modules

### PR DESCRIPTION
**Pull request by dweomer**
* remove glide files
* update Dockerfile
* update Makefile
* github.com/edgexfoundry/edgex-go/pkg -> github.com/edgexfoundry/go-mod-core-contracts
* models.DeviceObject -> models.DeviceResource

depends on edgexfoundry/ci-management#316

Please add appropriate reviewers @jamesrgregg @MightyNerdEric @ernestojeda 

Signed-off-by: Jacob Blain Christen <jacob.blain.christen@intel.com>

### CODE REVIEW

**Comment by dweomer** [comment]
addresses #26

**Comment by dweomer** [comment]
recheck

**Comment by dweomer** [comment]
Please review @SteveOss @tsconn23 @weichou1229 @cloudxxx8 
/cc @edgexfoundry/device-mqtt-go-committers

**Comment by dweomer** [comment]
Fixed a copy pasta typo.

File: Makefile [hunk 1]
```diff
@@ -16,23 +16,17 @@ GIT_SHA=$(shell git rev-parse HEAD)
...
 	$(GO) build $(GOFLAGS) -o $@ ./cmd
 
 test:
-	go test ./... -cover
+	$(GO) test ./... -cover
 
 clean:
 	rm -f $(MICROSERVICES)
 
-prepare:
```
Comment (tsconn23) [inline Makefile]: So the Jenkins job has been modified to not call this `make` step then?

**Comment by tsconn23** [comment]
LGTM other than question about the `make prepare` removal.

**Comment by dweomer** [comment]
> LGTM other than question about the `make prepare` removal.

The `prepare` target used to have glide establish the `vendor` directory but now that this project is using go modules it is superfluous (because dependency download is now rolled into `go build`), so I removed it.

It it should be put back in for convention sake I have no issue making that happen.

File: Makefile (hunk 1 above)
Comment (dweomer) [reply Makefile]: yes

**Comment by tsconn23** [comment]
@dweomer Understood, but make sure the JJB job isn't still calling it because if it is, the job will probably fail when the make target doesn't exist.

**Comment by dweomer** [comment]
@tsconn23 In an effort to honor the principle of least surprise I have put the prepare target back in the `Makefile`. It's a no-op at the moment but I do see reference to it in the job defined by ci-management (a den of witchery) and so let's just play it safe here, shall we?

**Comment by dweomer** [comment]
recheck

**Comment by cloudxxx8** [comment]
there must be a .go file of some sort in the root, otherwise "make test" will fail.
Since we are going to remove version.go, should we add another placeholder?

**Comment by dweomer** [comment]
@edgexfoundry/device-mqtt-go-committers all checks have passed

**Comment by jamesrgregg** [comment]
@SteveOss please assign reviewers thank you.

**Pull request merged**